src/esperanza/finalizationstate.cpp

```diff
@@ -1123,4 +1112,22 @@ bool FinalizationState::IsFinalizerVoting(const Validator &finalizer) const {
   return IsInDynasty(finalizer, m_current_dynasty) || IsInDynasty(finalizer, m_current_dynasty - 1);
 }
 
+uint32_t FinalizationState::CalculateWithdrawEpoch(const Validator &finalizer) const {
+
+  // m_end_dynasty is not set
+  if (finalizer.m_end_dynasty == DEFAULT_END_DYNASTY) {
```

Comment:
@frolosofsky the interface of `m_end_dynasty` wasn't changed in this PR. We set `DEFAULT_END_DYNASTY` if we don't have the actual value. 

But since it's not a big change, I can accommodate it here. I'll use `std::numeric_limits<uint32_t>::max()` everywhere instead of `DEFAULT_END_DYNASTY` as this value should always be max, `DEFAULT_END_DYNASTY` is not supposed to be changed to `2`

Reply:
> the interface of m_end_dynasty wasn't changed in this PR. We set DEFAULT_END_DYNASTY if we don't have the actual value.

It's true, but the way we apply the check is changed significantly. Currently in master we only compare `endDynasty < DEFAULT_END_DYNASTY`.

Reply:
Would not it be easier to express DEFAULT_END_DYNASTY as 0 maybe?

Reply:
No, because then you'll able to withdraw immediately  (if finalizer is in Genesis). 
Using the max allows us the returned value in every check and don't care about what is the value there.

Reply:
I think this `DEFAULT_` prefix confuses, we can call it `MAX_`. WDYS?

Reply:
MAX sounds fine. But please use it everywhere in the code (do not mix with `std::numeric_limits<uint32_t>::max()`)